### PR TITLE
Add POC advice2 app.

### PR DESCRIPTION
**Pull request by alixedi**

### CODE REVIEW

File: caseworker/advice2/apps.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class Advice2Config(AppConfig):
+    name = 'advice2'
```
Comment (saruniitr) [inline caseworker/advice2/apps.py:5]: `advice2` - any better alternatives? I have no suggestions to offer, maybe `advice_v2`?

File: caseworker/advice2/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"""ATM, all the caseworker unit tests that we have live in -
+
+unit_tests/caseworker/
+
+I would like us to have independent unit tests for advice2 and put them here.
```
Comment (saruniitr) [inline caseworker/advice2/tests/__init__.py:5]: any particular reason? why not keep in the existing place along with other tests? We are creating a new app for advice but tests can be in the same place I think unless there is a better reason?

File: caseworker/cases/urls.py [hunk 3]
```diff
@@ -13,6 +14,8 @@
     path("move/", main.MoveCase.as_view(), name="move"),
     path("additional-contacts/add/", main.AddAnAdditionalContact.as_view(), name="add_additional_contact"),
     path("attach/", main.AttachDocuments.as_view(), name="attach_documents"),
+    # This needs to be before "case" path b/c the regex in that sinks everything
+    path("advice2/", include("caseworker.advice2.urls")),
```
Comment (agalamatis) [inline caseworker/cases/urls.py:18]: Is there there an `/advice/` url used? Advice is currently all inside `/cases/`, so you can drop the `2` from the urls.

File: caseworker/advice2/apps.py (hunk 1 above)
Comment (alixedi) [reply caseworker/advice2/apps.py:5]: Sure. 

Or, like Ares suggested above, we can go with `advice` b/c there is no such app in here ATM.

Comment (saruniitr) [reply caseworker/advice2/apps.py:5]: :+1:

File: caseworker/advice2/tests/__init__.py (hunk 2 above)
Comment (saruniitr) [reply caseworker/advice2/tests/__init__.py:5]: that is a better reason, :+1:

File: caseworker/advice2/templates/advice2/placeholder.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{% extends 'layouts/case.html' %}
+
+{% load static custom_tags %}
+
+{% block body %}
+{{ greetings }}
+{% endblock %}
```
Comment (Pithikos) [inline caseworker/advice2/templates/advice2/placeholder.html:7]: @alixedi why the double-nesting of this file? Why not have caseworker/advice2/templates/placeholder.html ?

File: caseworker/advice2/tests/__init__.py (hunk 2 above)
Comment (Pithikos) [reply caseworker/advice2/tests/__init__.py:5]: Django has some default setup but we don't really need to abide by it (and some of their decisions actually don't work in many projects). We can either have tests per app or in a global directory - it's up to us

File: caseworker/advice2/views.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+from django.views.generic import TemplateView
+from caseworker.cases.services import get_case
+
+
+class Advice2PlaceholderView(TemplateView):
+    """This is POC ATM and should be removed with the first PR
```
Comment (Pithikos) [inline caseworker/advice2/views.py:6]: Can we please reduce abbreviations and just use lowercase for common day abbrevs like atm? :) Not sure what POC is..

File: caseworker/advice2/views.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    of advice2. Currently, this is a TemplateView but it should
+    be fairly simple to make this e.g. a SingleFormView.
+    """
+    template_name = "advice2/placeholder.html"
```
Comment (Pithikos) [inline caseworker/advice2/views.py:10]: Related to my comment on nesting.. Maybe we can remove the double `advice2` directory? So we could use `template_name = "placeholder.html"` here

File: caseworker/cases/urls.py (hunk 3 above)
Comment (dgnball) [reply caseworker/cases/urls.py:18]: agreed

**Comment by dgnball** [comment]
Would like to see an example of what using crispy forms within this app would look like.

File: caseworker/advice2/tests/__init__.py (hunk 2 above)
Comment (dgnball) [reply caseworker/advice2/tests/__init__.py:5]: Doesn't seem to be much value in moving the unit tests near to the code that is being tested over the current layout. I would advocate for consistency with what we are already doing in this case and put a sample test_views.py test in the existing unit_tests folder.

File: caseworker/advice2/templates/advice2/placeholder.html (hunk 4 above)
Comment (alixedi) [reply caseworker/advice2/templates/advice2/placeholder.html:7]: It is a Django convention. You can find out the reasons in the [Django tutorial](https://docs.djangoproject.com/en/dev/intro/reusable-apps/).

**Comment by dgnball** [comment]
Would also like to see an example of a form making an API call.

File: caseworker/advice2/tests/__init__.py (hunk 2 above)
Comment (alixedi) [reply caseworker/advice2/tests/__init__.py:5]: Yes, it's up to us and we can choose to deviate from conventions if we understand why the convention is there and weigh up the benefits of moving away from it.

Basically, the Django convention is for apps to be re-usable packages. Personally, even when I don't plan to re-use apps, designing them to be re-usable means that I tend to be more careful about the interfaces. This results in better code and is easier to test and reason about.

File: caseworker/advice2/views.py (hunk 5 above)
Comment (alixedi) [reply caseworker/advice2/views.py:6]: Sorry, POC is proof of concept.

File: caseworker/advice2/templates/advice2/placeholder.html (hunk 4 above)
Comment (dgnball) [reply caseworker/advice2/templates/advice2/placeholder.html:7]: I'm guessing this is Django style but I'd like to see a reduction in file nested if possible where possible

Comment (alixedi) [reply caseworker/advice2/templates/advice2/placeholder.html:7]: To give a more concrete example for this - If we had to override templates from e.g. `caseworker/cases` app in the `advice2` app, what directory structure would you need?

File: caseworker/advice2/templates/advice2/placeholder.html [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{% extends 'layouts/case.html' %}
```
Comment (eadpearce) [inline caseworker/advice2/templates/advice2/placeholder.html:1]: in caseworker/templates/layouts/case.html we can add a new block that wraps the `{% if tabs %}` chunk on line 118 and then override that in here

File: caseworker/advice2/templates/advice2/placeholder.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+{% load static custom_tags %}
+
+{% block body %}
```
Comment (eadpearce) [inline caseworker/advice2/templates/advice2/placeholder.html:5]: then if we do that we can put this content in `{% block two_thirds %}` instead

File: caseworker/advice2/templates/advice2/placeholder.html (hunk 4 above)
Comment (Pithikos) [reply caseworker/advice2/templates/advice2/placeholder.html:7]: OK, fair enough.. so it's to avoid name collision

**Review by eadpearce — APPROVED** [review]

**Pull request merged**